### PR TITLE
[WIP] Fix(kclvm-parser): Added 'lookbehind' method in lexer.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #65.

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm_lexer/parser

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

Added 'lookbehind' method in lexer.
For code containing unicode characters, find unicode characters in advance through lookbehind and report an error.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: kclvm/parser/src/lexer/mod.rs [hunk 1]
```diff
@@ -420,6 +434,33 @@ impl<'a> Lexer<'a> {
...
+    // From the lexed tokens stack, check whether the token at the top of the stack and the current character can combine a new token.
+    // If yes, lexer will pop the token at the top of the stack and return a new token combined with the token poped and the current character.
+    // If not, return None.
+    fn look_behind(
+        &mut self,
+        tok: &kclvm_lexer::Token,
+        tok_stream_builder: &mut TokenStreamBuilder,
+    ) -> Option<TokenKind> {
+        match tok.kind {
+            kclvm_lexer::TokenKind::Gt => {
```
Comment (Peefy) [inline kclvm/parser/src/lexer/mod.rs:446]: Is there only '->' case?

File: kclvm/parser/src/lexer/mod.rs [hunk 2]
```diff
@@ -687,6 +728,29 @@ impl TokenStreamBuilder {
         self.buf.push(token)
     }
 
+    // Pop the token at the top of the stack, and return None if the stack is empty.
+    fn pop(&mut self) -> Option<Token> {
+        self.buf.pop()
+    }
+
+    // If the token kind at the top of the stack is 'expected_tok_kind',
```
Comment (Peefy) [inline kclvm/parser/src/lexer/mod.rs:736]: s/`//`/`///`

File: kclvm/parser/src/lexer/mod.rs [hunk 3]
```diff
@@ -687,6 +728,29 @@ impl TokenStreamBuilder {
...
+    // pop the token and return it, otherwise do nothing and return None.
+    fn pop_if_tok_kind(&mut self, expected_tok_kind: &TokenKind) -> Option<Token> {
```
Comment (Peefy) [inline kclvm/parser/src/lexer/mod.rs:738]: s/`//`/`///`

File: kclvm/parser/src/lexer/mod.rs [hunk 4]
```diff
@@ -687,6 +728,29 @@ impl TokenStreamBuilder {
         self.buf.push(token)
     }
 
+    // Pop the token at the top of the stack, and return None if the stack is empty.
```
[the remark this answers is not in the document]
Comment (zong-zhe) [reply kclvm/parser/src/lexer/mod.rs]: fixed

File: kclvm/parser/src/lexer/mod.rs (hunk 2 above)
Comment (zong-zhe) [reply kclvm/parser/src/lexer/mod.rs]: fixed

File: kclvm/parser/src/lexer/mod.rs (hunk 3 above)
Comment (zong-zhe) [reply kclvm/parser/src/lexer/mod.rs:741]: fixed

File: kclvm/parser/src/lexer/mod.rs [hunk 5]
```diff
@@ -687,6 +728,29 @@ impl TokenStreamBuilder {
...
+    // pop the token and return it, otherwise do nothing and return None.
+    fn pop_if_tok_kind(&mut self, expected_tok_kind: &TokenKind) -> Option<Token> {
+        if self.peek_tok_kind() == expected_tok_kind {
+            self.pop()
+        } else {
+            None
+        }
+    }
+
+    // Peek the kind of the token on the top of the stack.
```
[the remark this answers is not in the document]
Comment (zong-zhe) [reply kclvm/parser/src/lexer/mod.rs]: fixed

File: kclvm/parser/src/lexer/tests.rs [hunk 6]
```diff
@@ -6,18 +6,42 @@ use kclvm_span::{create_session_globals_then, BytePos, FilePathMapping, SourceMa
 use std::path::PathBuf;
 use std::sync::Arc;
 
-fn check_lexing(src: &str, expect: Expect) {
+// lexing the 'src'.
+fn lex(src: &str) -> String {
```
[the remark this answers is not in the document]
Comment (zong-zhe) [reply kclvm/parser/src/lexer/tests.rs:10]: fixed

File: kclvm/parser/src/lexer/mod.rs (hunk 1 above)
Comment (zong-zhe) [reply kclvm/parser/src/lexer/mod.rs:449]: Added some comments for this case.

**Comment by ldxdl** [comment]
Pls consider & clarify:
+ E2E perf impact
+ remove look_bebind func wrapper?
+ add more unicode test cases

**Pull request closed without merging**